plugin/src/main/java/com/playmonumenta/scriptedquests/zones/zone/Zone.java

```diff
@@ -0,0 +1,273 @@
+import java.util.Set;
+import java.util.Map.Entry;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import org.bukkit.Axis;
+import org.bukkit.util.Vector;
+
+import com.playmonumenta.scriptedquests.zones.ZoneLayer;
+
+/*
+ * A fragment of a zone; this is used to find zones quickly, but not hold their properties.
+ * Instead, each fragment points to its parent, a zone with properties.
+ * Each zone also keeps track of its fragments.
+ */
+public class Zone extends BaseZone {
+	private ZoneLayer mLayer;
+	private String mName;
+	private ArrayList<ZoneFragment> mFragments = new ArrayList<ZoneFragment>();
+	private LinkedHashSet<String> mProperties = new LinkedHashSet<String>();
+
+	/*
+	 * Why yes, this does look weird. super() can only be called as the first method, and
+	 * the json needs to be parsed first - but we don't want to duplicate code from the
+	 * base class. By making this a static method, we made it possible to parse the
+	 * json before handling anything else, including super(). This looks doubly weird
+	 * for using the same method name - Zone - as the constructor, but this is a
+	 * constructor like any other.
+	 */
+	public static Zone ConstructFromJson(ZoneLayer layer, JsonObject object, HashMap<String, ArrayList<String>> propertyGroups) throws Exception {
+		if (layer == null) {
+			throw new Exception("layer may not be null.");
+		}
+		if (object == null) {
+			throw new Exception("object may not be null.");
+		}
+		if (propertyGroups == null) {
+			throw new Exception("propertyGroups may not be null (but may be empty).");
+		}
+
+		Double[] corners = new Double[6];
+		String name;
+		LinkedHashSet<String> properties = new LinkedHashSet<String>();
+
+		// Load the zone name
+		if (object.get("name") == null ||
+		    object.get("name").getAsString() == null ||
+		    object.get("name").getAsString().isEmpty()) {
+			throw new Exception("Failed to parse 'name'");
+		}
+		name = object.get("name").getAsString();
+
+		// Load the zone location
+		if (object.get("location") == null ||
+		    object.get("location").getAsJsonObject() == null) {
+			throw new Exception("Failed to parse 'location'");
+		}
+		JsonObject locationJson = object.get("location").getAsJsonObject();
+		Set<Entry<String, JsonElement>> entries = locationJson.entrySet();
+		for (Entry<String, JsonElement> ent : entries) {
+			String key = ent.getKey();
+			JsonElement value = ent.getValue();
+			switch (key) {
+			case "x1":
+				corners[0] = value.getAsDouble();
+				break;
+			case "y1":
+				corners[1] = value.getAsDouble();
+				break;
+			case "z1":
+				corners[2] = value.getAsDouble();
+				break;
+			case "x2":
+				corners[3] = value.getAsDouble();
+				break;
+			case "y2":
+				corners[4] = value.getAsDouble();
+				break;
+			case "z2":
+				corners[5] = value.getAsDouble();
+				break;
+			default:
+				throw new Exception("Unknown location key: '" + key + "'");
+			}
+		}
```

Comment:
We should really move this to some type of "area" object... we seem to do this a lot. Doesn't have to be done in this change, just a thought.